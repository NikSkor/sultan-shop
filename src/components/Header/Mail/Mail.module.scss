.block {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 15px;
}

.info {
  display: flex;
  flex-direction: column;
}

.title {
  font-weight: 600;
  font-size: 14px;
  line-height: 18px;
  color: inherit;
}

.description {
  font-weight: 300;
  font-size: 12px;
  line-height: 130.02%;
}