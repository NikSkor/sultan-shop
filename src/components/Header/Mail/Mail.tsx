import React, { FC } from 'react';
import style from './Mail.module.scss';

interface MailProps {
  href: string,
  description: string,
  children?: React.ReactNode,
  titleColor: string,
  descrColor: string,
}

const Mail: FC<MailProps> = ({href, description, titleColor, descrColor, children}) => {
  return (
    <div className={style.block}>
      {children}
      <div className={style.info}>
        <a href={`mailto:${href}`} className={style.title} style={{color: titleColor}}>{href}</a>
        <div className={style.description} style={{color: descrColor}}>{description}</div>
      </div>
    </div>
  )
}

export default Mail;