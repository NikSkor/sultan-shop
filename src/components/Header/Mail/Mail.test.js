import { cleanup, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Mail from '../Mail/Mail.tsx';
import letterImg from '../../../img/letter.png';


describe('Mail.tsx', () => {
afterEach(cleanup);

  test('component renders props correctly, truthy values', () => {
    const tree = renderer
      .create(
        <Mail
          isHeader={true}
          href='opt.sultan@mail.ru'
          description='На связи в любое время'
          titleColor='#111111'
          descrColor='#3F4E65'
        >
          <img src={letterImg} alt='Значок почты' />
        </Mail>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('component renders props correctly, falsy values', () => {
    const tree = renderer.create(<Mail />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('href role', ()=> {
    render(<Mail />);
    const link = screen.getByRole('link');
    expect(link).toBeInTheDocument();
    expect(link).toMatchSnapshot();
  })
});
