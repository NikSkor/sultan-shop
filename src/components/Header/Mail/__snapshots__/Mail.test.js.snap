// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mail.tsx component renders props correctly, falsy values 1`] = `
<div
  className="block"
>
  <div
    className="info"
  >
    <a
      className="title"
      href="mailto:undefined"
      style={
        Object {
          "color": undefined,
        }
      }
    />
    <div
      className="description"
      style={
        Object {
          "color": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Mail.tsx component renders props correctly, truthy values 1`] = `
<div
  className="block blockHeader"
>
  <img
    alt="Значок почты"
    src="letter.png"
  />
  <div
    className="info"
  >
    <a
      className="title"
      href="mailto:opt.sultan@mail.ru"
      style={
        Object {
          "color": "#111111",
        }
      }
    >
      opt.sultan@mail.ru
    </a>
    <div
      className="description"
      style={
        Object {
          "color": "#3F4E65",
        }
      }
    >
      На связи в любое время
    </div>
  </div>
</div>
`;

exports[`Mail.tsx href role 1`] = `
<a
  class="title"
  href="mailto:undefined"
/>
`;
